integration/kotlinx-coroutines-play-services/src/Tasks.kt

```diff
@@ -45,29 +38,56 @@ public fun <T> Deferred<T>.asTask(): Task<T> {
 /**
  * Converts this task to an instance of [Deferred].
  * If task is cancelled then resulting deferred will be cancelled as well.
+ * For bi-directional cancellation, an overload that accepts [CancellationTokenSource] can be used.
  */
-public fun <T> Task<T>.asDeferred(): Deferred<T> {
+public fun <T> Task<T>.asDeferred(): Deferred<T> = asDeferredImpl(null)
+
+/**
+ * Converts this task to an instance of [Deferred] with a [CancellationTokenSource] to control cancellation.
+ * The cancellation of this function is bi-directional:
+ * If the given task is cancelled, the resulting deferred will be cancelled.
+ * If the resulting deferred is cancelled, the provided [cancellationTokenSource] will be cancelled.
```

Comment:
```suggestion
 * * If the given task is cancelled, the resulting deferred will be cancelled.
 * * If the resulting deferred is cancelled, the provided [cancellationTokenSource] will be cancelled.
```
The IDE ignores newlines when formatting doc comments, so the layout looks confusing otherwise.

Reply:
Good catch!